### PR TITLE
Fix race condition between ProcessList and Settings

### PR DESCRIPTION
**Pull request by alexey-milovidov**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

See https://s3.amazonaws.com/clickhouse-test-reports/0/1d8e886d7b58dd56ecdd6761ad5673d1838b347d/stateless_tests__tsan__s3_storage__[1_4]/stderr.log

Appeared after https://github.com/ClickHouse/ClickHouse/pull/66912

It happens when a query modifies its own settings dynamically, and at the same time, someone looks at system.processes.

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [ ]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

File: src/Interpreters/ProcessList.cpp [hunk 1]
```diff
@@ -657,7 +657,7 @@ QueryStatusInfo QueryStatus::getInfo(bool get_thread_list, bool get_profile_even
     {
         if (auto ctx = context.lock())
         {
-            res.query_settings = std::make_shared<Settings>(ctx->getSettingsRef());
+            res.query_settings = std::make_shared<Settings>(ctx->getSettingsCopy());
```
Comment (alexey-milovidov) [inline src/Interpreters/ProcessList.cpp:660]: Only this line. All others are misc changes.

**Comment by robot-ch-test-poll3** [comment]
*This is an automated comment for commit 69bd306a445a6bc8a55be14bb0080864921f8b69 with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/67645/69bd306a445a6bc8a55be14bb0080864921f8b69/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td>✅ success</td></tr>
<tbody>
</table>
</details>

**Review by pufit — APPROVED** [review]

**Comment by alexey-milovidov** [comment]
https://github.com/ClickHouse/ClickHouse/issues/65166

**Comment by alexey-milovidov** [comment]
Needs #57728

**Pull request merged**